### ISSUE TITLE
[BUG] SwProductSelect doesn't render itself based on product variants

### ISSUE THREAD

**Issue by Radek-Wawrzyk** [DivanteLtd/shopware-pwa#1187]
### Describe the bug
When I add a new product variants, the **SwProductSelect** component doesn't render these variants (**SwProductDetails.vue** component). Also computed getters **getAllProductOptions, hasChildren and getAllProductOptionsTypes** are empty. What is interesting, is that the default shopware6 shop has these variants to choose (screenshot below in the Additional context section). I suppose that they should render itself by default.

### Steps to reproduce

1. Create a new product in the shopware6 admin dashboard
2. Create few variants of the main product
3. Go to that product page (PWA shopware app) and check the product object and computed properties (SwProductDetails.vue component)
4. Go to the default shopware6 shop, find that product and check the variants (they should be visible as on screenshot below)

### Additional context
**Screenshot of computed properties (PWA shopware)**

![Screenshot from 2020-10-19 16-03-40](https://user-images.githubusercontent.com/25388307/96461508-c1887400-1224-11eb-8d2d-3cfeed9aa3ac.png)

**Screenshot of default shopware6 shop (not PWA)**

![Screenshot from 2020-10-19 15-44-21](https://user-images.githubusercontent.com/25388307/96461590-db29bb80-1224-11eb-8f65-e79dbc2e4a70.png)

Shopware6 PWA version: **^0.4.2**
Shopware6 Dashboard version: **6.3.0.1**

**Comment by mkucmus** [comment]
hi @Radek-Wawrzyk, thanks for reporting the issue :+1: 

I have created a couple of variants for a specific product, the result is here: https://shopware-pwa.storefrontcloud.io/detail/fb14584a96864e1ba607e04d41625a84

shopware: 6.3.0.1
pwa: latest stable

so maybe there are missing parts within the request itself :thinking: 

in order to reproduce the entire env please paste the request you are sending while you are entering the product details page (CURL format preferred) :muscle: and the response maybe?
we will figure it out :)

**Comment by Radek-Wawrzyk** [comment]
### CURL request
```
curl 'http://admin.britishstandard.test:8108/store-api/v3/pwa/page' \
  -H 'Connection: keep-alive' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'sw-access-key: SWSCRTZCDM5TEWE0EDLSZMRVTQ' \
  -H 'sw-context-token: T0N0mFSLuo7xZW7g2HNL5CHgWIoC5xyL' \
  -H 'sw-language-id: 2fbb5fe2e29a4d70aa5854ce7ce3e20b' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.102 Safari/537.36' \
  -H 'Content-Type: application/json;charset=UTF-8' \
  -H 'Origin: http://britishstandard.test:8108' \
  -H 'Referer: http://britishstandard.test:8108/' \
  -H 'Accept-Language: pl,en-GB;q=0.9,en;q=0.8,pl-PL;q=0.7,en-US;q=0.6,la;q=0.5' \
  --data-binary '{"path":"detail/1aac798959784c769b73d5758df0b7a8","limit":10,"associations":{"media":{},"properties":{"associations":{"group":{}}},"productReviews":{},"children":{"associations":{"options":{"associations":{"group":{}}}}},"options":{"associations":{"group":{}}}},"includes":{"cms_page_slot":["id","type","slot","blockId","config","data","backgroundMediaMode","backgroundMedia"],"cms_page_block":["slots","type","id","backgroundColor","backgroundMedia","sectionPosition","backgroundColor"],"cms_page_section":["id","backgroundMedia","blocks","type","sizingMode"],"cms_page":["id","name","sections","type","config"],"product":["manufacturer","id","name","properties","deliveryTime","stock","media","productNumber","children","calculatedListingPrice","translated","options","properties","manufacturer"],"product_media":["media"],"media":["thumbnails","width","height","url"],"calculated_price":["unitPrice","quantity"],"product_group_option":["name","id","group","translated"],"product_group":["id","name","options","translated"],"property_group":["id","translated","options"],"property_group_option":["name","translated","id","colorHexCode","media","group"]}}' \
  --compressed \
  --insecure
```

### shopware-pwa.config.js

```
module.exports = {
  apiDefaults: {
    useCms: {
      includes: {
        product: [
          "manufacturer", "id", "name", "properties", "deliveryTime",
          "stock", "media", "productNumber", "children", "calculatedListingPrice"
        ],
        property_group: ['id', 'translated', 'options'],
        property_group_option: [
          'name',
          'translated',
          'id',
          'colorHexCode',
          'media',
          'group',
        ],
      },
      associations: [
        { name: "media" },
      ]
    },
  },
};
```

**Comment by mkucmus** [comment]
missing variants may be caused by lack of associations/includes params within the request (loadAssociations). will be fixed in milestone #v0.5.1

**Comment by mkucmus** [comment]
@Radek-Wawrzyk have you checked it on v0.5.1?

**Comment by mkucmus** [comment]
it has been fixed in 0.5.2
